### PR TITLE
Convert a bunch of legacy overlay calls to SSoverlays

### PR DESCRIPTION
**Pull request by Lohikar**
Bunch of overlay conversions I found in an old stash I had forgotten about.

DNM: Needs testing.

### CODE REVIEW

File: code/modules/vehicles/cargo_train.dm [hunk 1]
```diff
@@ -302,24 +302,20 @@
 	C.forceMove(src)
 
 	if(load_item_visible)
-		C.pixel_x += load_offset_x
-		C.pixel_y += load_offset_y
-		C.layer = layer
+		var/mutable_appearance/MA = new(C)
+		MA.pixel_x += load_offset_x
```
Comment (Lohikar) [inline code/modules/vehicles/cargo_train.dm]: and here.

File: code/modules/reagents/reagent_dispenser.dm [hunk 2]
```diff
@@ -131,12 +131,12 @@
...
 			user.drop_item()
-			W.loc = src
+			W.forceMove(src)
 
-			var/icon/test = getFlatIcon(W)
-			test.Shift(NORTH,1)
-			test.Shift(EAST,6)
-			overlays += test
+			var/mutable_appearance/MA = new(W)
+			MA.pixel_x += 1
```
Comment (Lohikar) [inline code/modules/reagents/reagent_dispenser.dm]: Can't remember if pixel offsets work on overlays. Needs testing.

File: code/modules/mob/living/silicon/robot/gripper.dm [hunk 3]
```diff
@@ -75,23 +75,13 @@
...
 /obj/item/weapon/gripper/update_icon()
-	overlays.Cut()//Dummy object will be picked up by garbage collection
+	underlays.Cut()
 	if (wrapped && wrapped.icon)
+		var/mutable_appearance/MA = new(wrapped)
+		MA.layer = FLOAT_LAYER
+		MA.pixel_y = -8
 
-
-		var/obj/dummy = new //We create a dummy object which looks like the gripped item, to use as an overlay
```
Comment (Lohikar) [inline code/modules/mob/living/silicon/robot/gripper.dm]: It hurts.

File: code/controllers/subsystems/overlays.dm [hunk 4]
```diff
@@ -185,6 +185,24 @@ var/datum/controller/subsystem/overlays/SSoverlays
...
 
+/atom/proc/set_overlays(list/overlays, priority = FALSE)	// Sets overlays to a list, equivalent to cut_overlays() + add_overlays().
+	if (!overlays)
+		return
+
+	overlays = build_appearance_list(overlays)
+
+	if (priority)
+		LAZYCLEARLIST(priority_overlays)
+		if (overlays)
```
Comment (skull132) [inline code/controllers/subsystems/overlays.dm]: Can `build_appearance_list` return null? This is the only case where this check would be useful.

File: code/game/objects/effects/decals/crayon.dm [hunk 5]
```diff
@@ -5,26 +5,30 @@
...
 		shadeOverlay.Blend(shade,ICON_ADD)
+		SSicon_cache.crayon_cache["[type]_s"] = shadeOverlay
 
-		overlays += mainOverlay
-		overlays += shadeOverlay
+	add_overlay(list(mainOverlay, shadeOverlay))
 
-		add_hiddenprint(usr)
\ No newline at end of file
+	add_hiddenprint(usr)
```
Comment (skull132) [inline code/game/objects/effects/decals/crayon.dm]: hnrg old code plz.

**Pull request merged**